### PR TITLE
Add IAP core code

### PR DESCRIPTION
**Pull request by jantimar**
Add InAppPurchases code
Add OpenSSL dependence

### CODE REVIEW

File: Sources/youda/ASN1/ASN1Reader+Purchase.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+//  MIT license, see LICENSE file for details
+//
+
+import Foundation
+import OpenSSL
+
+extension ASN1Reader {
+  func readPurchase(
+    payloadLength: Int
+  ) throws -> Purchase {
```
Comment (VojtaStavik) [inline Sources/youda/ASN1/ASN1Reader+Purchase.swift]: Do we really have to have the function signature spread over 3 lines?

File: Sources/youda/ReceiptService.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  /// Verify Receipt SHA1 hash
+  /// - Parameter receipt: receipt for verification
+  @available(iOS 13.0, *)
+  public func verify(receipt: Receipt) throws {
+    guard
+      let opaque = receipt.opaque,
+      let bundleID = receipt.bundleIdRawData,
+      let hash = receipt.hash,
+      var deviceID = deviceID?.uuid else {
```
Comment (VojtaStavik) [inline Sources/youda/ReceiptService.swift:46]: `else {` should be on a new line

File: Sources/youda/InAppPurchases.swift [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+//
+
+import Foundation
+import StoreKit
+
+public struct InAppProductId: Hashable {
+  let identifier: String
+}
+
+public let subscriptionChangeNotification: Notification.Name = Notification.Name(rawValue: "io.insdustrial-binaries.subscription-change")
```
Comment (VojtaStavik) [inline Sources/youda/InAppPurchases.swift]: 1) It's usually better to declare a notification name like this:
```
extension Notification.Name { 
  static var subscriptionChange: Notification.Name { ... }
}
```

2) Typos in the name. Maybe should be `co.industrial-binaries.youda.subscription-change`?

File: Sources/youda/ASN1/ASN1Reader.swift [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+//  youda
+//
+//  Copyright (c) 2020 Industrial Binaries
+//  MIT license, see LICENSE file for details
+//
+
+import Foundation
+import OpenSSL
+
+final class ASN1Reader {
```
Comment (VojtaStavik) [inline Sources/youda/ASN1/ASN1Reader.swift:11]: We can eventually extract this into a separate package. Let's not worry about that now.

File: Sources/youda/Certificate/CertificateService.swift [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import OpenSSL
+import StoreKit
+
+/// * The Apple root certificate, as raw data and in its OpenSSL representation. */
+// BIO *b_x509;
+// X509 *Apple;
+struct CertificateService {
+  enum Certificate: String {
+		/// Apple Inc Root Certificate for verify Receipt from https://www.apple.com/certificateauthority/
+    case appleRoot = "AppleIncRootCertificate"
```
Comment (VojtaStavik) [inline Sources/youda/Certificate/CertificateService.swift]: Why we have an `enum` with one `case`?

**Review by VojtaStavik — APPROVED** [review]
I did just basic review. The crypto stuff is not my domain. btw - if we use snippets from someone else it would be good to always add a link to the source.

File: Sources/youda/ASN1/ASN1Reader+Purchase.swift (hunk 1 above)
Comment (jantimar) [reply Sources/youda/ASN1/ASN1Reader+Purchase.swift]: No, my fault

**Pull request merged**